### ISSUE TITLE
Getting Cython-related error when running Python sampler

### ISSUE THREAD

**Issue by eggplantbren** [eggplantbren/DNest4#17]
Hi Dan,

Am I doing something stupid? Was just trying to run the `gaussian.py` example and it's complaining that I didn't build the Cython extensions, even though I did.

`python gaussian.py 
Traceback (most recent call last):
  File "gaussian.py", line 43, in <module>
    sep=" "))
  File "/home/brendon/local/anaconda3/lib/python3.5/site-packages/dnest4-0.1.4-py3.5-linux-x86_64.egg/dnest4/sampler.py", line 17, in __init__
    raise ImportError("You must build the Cython extensions to use "
ImportError: You must build the Cython extensions to use the Python API`

**Comment by eggplantbren** [comment]
I'm using Anaconda 4.2.0 on Ubuntu 16.10, and the DNest python package was successfully installed with `python setup.py install`. Let me know what other info might help...

**Comment by MichelleLochner** [comment]
@eggplantbren I had the same problem with anaconda. It turns out to be a mismatch between the compiler used to compile DNest4 and the compiler used by anaconda when running cythonize. I did the following (after making DNest4):

`export CC=/usr/bin/gcc`
`export CXX=/usr/bin/g++`
`python setup.py install`

I'm not sure if setting both compilers is necessary or just the C++ one but after this it worked.

**Comment by eggplantbren** [comment]
I'm glad this worked for you, but it didn't for me...😕

**Comment by eggplantbren** [comment]
This was a strange issue with my setup that seems to have been resolved. Closing

**Comment by eggplantbren** [comment]
This is happening again on two separate fresh installs of Anaconda 5.3.1 for Linux. I have no clue why - I have Cython and I saw it compiling when I installed the package. I've also tried Michelle's suggestion, which didn't help. Let me know of anything that might diagnose this, @dfm. It's a fairly severe problem if this happens to a user.

**Comment by dfm** [comment]
This normally has to do with trying to run within the directory where the package is installed. It can help to use: `python setup.py develop` instead of `install`. You might also try running `pip uninstall dnest4` a few times first. Hope this helps!

**Comment by eggplantbren** [comment]
I just tried all of those suggestions and none of them worked.

**Comment by dfm** [comment]
What happens when you run:

```
from dnest4._dnest4 import sample
```

from Python?

**Comment by eggplantbren** [comment]
I wasn't expecting this, but it seems like a clue!

```
Python 3.7.0 (default, Oct  9 2018, 10:31:47) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dnest4
>>> from dnest4._dnest4 import sample
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: /home/brewer/local/anaconda3/lib/python3.7/site-packages/dnest4-0.2.3-py3.7-linux-x86_64.egg/dnest4/_dnest4.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZN6DNest46CauchyC1Edd
```

**Comment by dfm** [comment]
Looks like there might be new files that need to be included. Take a look at setup.py in the Python directory and see which files are missing? I'm not sure where Cauchy is defined.

**Comment by eggplantbren** [comment]
That was exactly it. I just didn't know I had to edit setup.py after adding more features in the C++, but now I do. Thanks heaps. Closed with commit b76065322733a48a34eeca32ade23f3d2750df3c

**Comment by dfm** [comment]
Excellent.
